rocketpy/simulation/flight.py

```diff
@@ -1860,28 +1818,7 @@ def u_dot_generalized(self, t, u, post_processing=False):
         u_dot = [*r_dot, *v_dot, *e_dot, *w_dot]
 
         if post_processing:
-            # Dynamics variables
-            self.R1_list.append([t, R1])
-            self.R2_list.append([t, R2])
-            self.R3_list.append([t, R3])
-            self.M1_list.append([t, M1])
-            self.M2_list.append([t, M2])
-            self.M3_list.append([t, M3])
-            # Atmospheric Conditions
-            self.wind_velocity_x_list.append(
-                [t, self.env.wind_velocity_x.get_value_opt(z)]
-            )
-            self.wind_velocity_y_list.append(
-                [t, self.env.wind_velocity_y.get_value_opt(z)]
-            )
-            self.density_list.append([t, self.env.density.get_value_opt(z)])
-            self.dynamic_viscosity_list.append(
-                [t, self.env.dynamic_viscosity.get_value_opt(z)]
-            )
-            self.pressure_list.append([t, self.env.pressure.get_value_opt(z)])
-            self.speed_of_sound_list.append(
-                [t, self.env.speed_of_sound.get_value_opt(z)]
-            )
+            u_dot.extend([R1, R2, R3, M1, M2, M3])
```

Comment:
I disagree with the way this is implemented. Changing the return of the udot methods is confusing since it needs to return only the states for its use in the integrator. 

I think appending a list here would be cleaner and faster. However, differently from the old code, it should be just one list similar to what you have done here

Reply:
Understood.

After my changes, every time we call the `u_dot` with `post_process=True`, the code is doing 1 extend, some replacements, but no appending. Your suggestion is to do 1 append only.

On the tip of my head I can't really tell which one is faster. Although your suggestion seems faster due to less operations,  I wonder if the extend could beat the append in terms of time.

I have to think/test, but I will probably accept your suggestion.

Reply:
Ok no problem, 

I pushed the [2013f16](https://github.com/RocketPy-Team/RocketPy/pull/581/commits/2013f1607b23548cdc309eedda51570097086d5e) commit with a few changes here.

There are "weak breaking changes". I'm no longer using the "R1_list" atributes in the Flight class.

I believe the code is cleaner now! But a few tests had to be changed due to the old acceleration initialization.